src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.File.Base.Unix.cs

```diff
@@ -51,5 +52,68 @@ protected void VerifyPlatformSpecificMetadata(string filePath, TarEntry entry)
                 }
             }
         }
+
+        protected int CreateGroup(string groupName)
+        {
+            int exitCode = Execute("groupadd", groupName, out string standardOutput, out string standardError);
+            if (exitCode != 0)
+            {
+                ThrowOnError(exitCode, "groupadd", groupName, standardError);
+            }
+            return GetGroupId(groupName);
+        }
+
+        protected int GetGroupId(string groupName)
+        {
+            int exitCode = Execute("getent", $"group {groupName}", out string standardOutput, out string standardError);
+            if (exitCode != 0)
+            {
+                ThrowOnError(exitCode, "getent", "group", standardError);
+            }
+
+            string[] values = standardOutput.Split(':', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+
+            return int.Parse(values[^1]);
+        }
+        
+        protected void SetGroupAsOwnerOfFile(string groupName, string filePath)
+        {
+            int exitCode = Execute("chgrp", $"{groupName} {filePath}", out string standardOutput, out string standardError);
+            if (exitCode != 0)
+            {
+                ThrowOnError(exitCode, "chgroup", $"{groupName} {filePath}", standardError);
+            }
+        }
+
+        protected void DeleteGroup(string groupName)
+        {
+            int exitCode = Execute("groupdel", groupName, out string standardOutput, out string standardError);
+            if (exitCode != 0)
+            {
+                ThrowOnError(exitCode, "groupdel", groupName, standardError);
+            }
+        }
+
+        private int Execute(string command, string arguments, out string standardOutput, out string standardError)
+        {
+            using Process p = new Process();
+
+            p.StartInfo.UseShellExecute = false;
+            p.StartInfo.FileName = command;
+            p.StartInfo.Arguments = arguments;
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.RedirectStandardError = true;
+            p.Start();
+            p.WaitForExit();
+
+            standardOutput = p.StandardOutput.ReadToEnd();
+            standardError = p.StandardError.ReadToEnd();
+            return p.ExitCode;
+        }
+
+        private void ThrowOnError(int code, string command, string arguments, string message)
+        {
+            throw new IOException($"Error '{code}' when executing '{command} {arguments}'. Message: {message}");
+        }
```

Comment:
all the callers of `Execute` implement the same pattern: check for the exit code and throw if it's not zero. We could move this logic to `Execute` itself and just simplify the usage

```suggestion
        protected int CreateGroup(string groupName)
        {
            Execute("groupadd", groupName);
            return GetGroupId(groupName);
        }

        protected int GetGroupId(string groupName)
        {
            string standardOutput = Execute("getent", $"group {groupName}");
            string[] values = standardOutput.Split(':', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);

            return int.Parse(values[^1]);
        }
        
        protected void SetGroupAsOwnerOfFile(string groupName, string filePath)
            => Execute("chgrp", $"{groupName} {filePath}");

        protected void DeleteGroup(string groupName)
            => Execute("groupdel", groupName);

        private string Execute(string command, string arguments)
        {
            using Process p = new Process();

            p.StartInfo.UseShellExecute = false;
            p.StartInfo.FileName = command;
            p.StartInfo.Arguments = arguments;
            p.StartInfo.RedirectStandardOutput = true;
            p.StartInfo.RedirectStandardError = true;
            p.Start();
            p.WaitForExit();

            string standardOutput = p.StandardOutput.ReadToEnd();
            string standardError = p.StandardError.ReadToEnd();
            
            if (p.ExitCode != 0)
            {
                throw new IOException($"Exit code '{p.ExitCode}' when executing '{command} {arguments}'. Standard error: {standardError}");
            }
            
            return standardOutput;
        }
```